NodeModuleCopyHelper excluding certain top-level files leads to broken binaries

[In this line of code](https://github.com/electron-userland/electron-builder/blob/562870e92dea50b2d20e4fd8519ac8f5293b6336/packages/app-builder-lib/src/util/NodeModuleCopyHelper.ts#L16), "test.js" is added to this list of `topLevelExcludedFiles` that are not copied over when building the binary. These files are intended to be READMEs, test files etc. that can be safely removed from production binaries, but this assumption doesn't always hold, and it doesn't seem possible to overwrite `topLevelExcludedFiles` anywhere.

For example, [Playwright 1.38.0](https://www.npmjs.com/package/playwright/v/1.38.0) added a top-level [test.js file](https://github.com/microsoft/playwright/blob/d6ec1ae3994f127e38b866a231a34efc6a4cac0d/packages/playwright/test.js#L1) which cannot be safely excluded. Without it, when you attempt to open the built binary, it fails to load with message

```
Error: Cannot find module '/Users/.../<appName>.app/Contents/Resources/app/node_modules/playwright/test.js'
at createEsmNotFoundErr (node:internal/modules/cjs/loader:990:15)
```

I confirmed this was the cause of the above behavior by modifying the `node_modules/app-builder/lib/out/util/NodeModuleCopyHelper.js` file myself and removing `test.js` from the list of excluded files, then running the build command, and seeing that the built app worked.

This is with Node 16, npm v8. Tested only on a Mac but I assume the behavior is replicable with other OS'es.

There does not seem to be a workaround by modifying the `files` argument: adding `node_modules/**/playwright/test.js` to the `files` globs still results in the file not being copied over, which is what the code seems to suggest would occur.

In the short term, maybe `test.js` is not a filename that can be assumed to be safely removable, so it could be removed from that list. It seems like the ideal fix is that these top-level excluded files are pre-emptible with the passed in `files` argument, rather than being hardcoded in and apparently impossible to override.

--- a/.changeset/cold-carpets-move.md
+++ b/.changeset/cold-carpets-move.md
@@ -1,0 +1,5 @@
+---
+"app-builder-lib": patch
+---
+
+Allowing `test.js` in compiled asar to allow testing mechanisms like Playwright

--- a/packages/app-builder-lib/src/util/NodeModuleCopyHelper.ts
+++ b/packages/app-builder-lib/src/util/NodeModuleCopyHelper.ts
@@ -13,7 +13,6 @@ const excludedFiles = new Set(
   )
 )
 const topLevelExcludedFiles = new Set([
-  "test.js",
   "karma.conf.js",
   ".coveralls.yml",
   "README.md",

